batch/test/test_batch.py

```diff
@@ -1056,3 +1056,99 @@ def test_job_private_instance_cancel(client: BatchClient):
     b.cancel()
     status = j.wait()
     assert status['state'] == 'Cancelled', str((status, b.debug_info()))
+
+
+def test_update_batch_no_deps(client: BatchClient):
+    b = client.create_batch()
+    b.create_job(DOCKER_ROOT_IMAGE, ['false'])
+    b = b.submit()
+
+    u = client.update_batch(b.id)
```

Comment:
With the changes we've made to the client, is it possible now to elide this line? Is there any scenario where after the  first `b.submit()`, we would not be in the `update` mode? I think of the batch client now as a buffer of jobs where `submit` acts like `flush`, so we could ideally write:

```python
b = client.create_batch()
b.create_job(DOCKER_ROOT_IMAGE, ['false'])
b.submit()
j = b.create_job(DOCKER_ROOT_IMAGE, ['true'])
b.submit()
status = j.wait()
```

Reply:
Sure! Good call.